File: examples/ruby/cond-write-multi-field/CondWriteMultiField.runar.rb
require 'runar'

# CondWriteMultiField -- regression fixture for GitHub issue #99: a
# conditional write of two mutable state fields in an `if` without an `else`.
class CondWriteMultiField < Runar::StatefulSmartContract
  prop :a, Bigint
  prop :b, Bigint

  def initialize(a, b)
    super(a, b)
    @a = a
    @b = b
  end

  runar_public flag: Bigint
  def bump(flag)
    if flag > 0
      @a = @a + 1
      @b = @b + 2
    end
    add_output(1000, @a, @b)
  end
end

File: examples/python/cond-write-multi-field/CondWriteMultiField.runar.py
from runar import (
    StatefulSmartContract, Bigint, public,
)


class CondWriteMultiField(StatefulSmartContract):
    """Regression fixture for GitHub issue #99 -- a conditional write of two
    mutable state fields in an `if` without an `else`."""

    a: Bigint
    b: Bigint

    def __init__(self, a: Bigint, b: Bigint):
        super().__init__(a, b)
        self.a = a
        self.b = b

    @public
    def bump(self, flag: Bigint):
        if flag > 0:
            self.a = self.a + 1
            self.b = self.b + 2
        self.add_output(1000, self.a, self.b)
